lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1132,12 +1133,27 @@ private ObjectData buildObjectData(
                     } else {
                         textValue = String.valueOf(objValue);
                     }
-                    b.withKeyProperty(key.substring(4), textValue);
+                    b.withKeyProperty(stripKeyPrefix(key), textValue);
                 }
             }
         }
+        if(pp.objectHighlight) {
+            for(String key : highlight.keySet()) {
```

Comment:
Make variable final.

This applies all over in this PR - always make variables final unless there's a good reason for them not to be.

Reply:
okay